src/cart/crt_hg.c

```diff
@@ -737,16 +737,37 @@ crt_get_info_string(bool primary, crt_provider_t provider, int iface_idx,
 	return rc;
 }
 
+static FILE    *hg_dbg_log        = NULL;
+static uint64_t last_hg_dbg_flush = 0;
+
 static int
 crt_hg_log(FILE *stream, const char *fmt, ...)
 {
 	va_list		ap;
+	struct timeval  tv;
 	int		flags;
 
 	flags = d_log_check((intptr_t)stream);
 	if (flags == 0)
 		return 0;
 
+	/*
+	 * b/401315986: If a separate logfile has been defined for
+	 * Mercury debug logging, send DEBUG-level messages to it.
+	 */
+	if (hg_dbg_log != NULL && (flags & DLOG_PRIMASK) == DLOG_DBG) {
+		va_start(ap, fmt);
+		vfprintf(hg_dbg_log, fmt, ap);
+		va_end(ap);
+
+		(void)gettimeofday(&tv, 0);
```

Comment:
I wonder if we should consider making this more generic.  In other words, what if we did this at the DAOS level (e.g D_LOG_DEBUG_FILE) ?  This would allow us to turn on debug generically without flooding log explorer if we so choose.

Otherwise, you lose the per message timestamp, no?

Reply:
Another advantage is that we buffer messages below warn